src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -79,23 +80,22 @@ internal class AssemblyLoader
             return discoveredAssemblyLocations;
         }
 #endif
-        public static T TryLoadAndCreateInstance<T>(string assemblyName, Logger logger) where T : class
+        public static T TryLoadAndCreateInstance<T>(string assemblyName, IServiceProvider serviceProvider, Logger logger) where T : class
```

Comment:
Suggestion - feel free to ignore, as this can be done later.
Suggest introducing a generic type factory interface and passing it in rather than the entire IServiceProvider.  Something like:

interface ITypeFactory
{
    T Create<T>(Type concreteType)  where T : class;
}

An impl of this that calls the activator utilities with the service provider can be passed in making this call path more specific and not depend on DI.

Again, this can be done as a minor refactor later..